src/top-navigation/styles.scss

```diff
@@ -58,6 +58,7 @@
 
 .identity {
   min-width: 0;
+  flex-grow: 1;
```

Comment:
@just-boris so you are right about this, but the problem is that we have magical calculations happens in the `useTopNavigation` that figures out the identity width, given the fact that flex-grow would make its width bigger than the top navigation container itself so I had to re-calculate the identity width below using its inner content ( logo width + title width ) in the virtual top navigation.

Kudos for the suggestion tho.

Reply:
Thanks for the extra dive. This approach is better!